sbi/neural_nets/density_estimators/base.py

```diff
@@ -48,11 +72,66 @@ def loss(self, input: Tensor, condition: Tensor, **kwargs) -> Tensor:
     def sample(self, sample_shape: torch.Size, condition: Tensor, **kwargs) -> Tensor:
         r"""Return samples from the density estimator.
 
+        Note:
+            This function should support batched conditions and should admit the
+            following behavior for different condition shapes:
+            - (*condition_shape) -> (*sample_shape,d)
+            - (*batch_shapes, *condition_shape) -> (*batch_shapes, *sample_shape, d)
```

Comment:
what is `batch_shapes`? if it is the number of passed conditions, we could call it `condition_batch_size`, or `num_conditions`, or `condition_batch`?

Reply:
Ups, sorry, it is an inconsistency should be batch_shape (not plural).

But best would probably be  `condition_batch_shape`